test/e2e/framework/syncer.go

```diff
@@ -511,6 +515,256 @@ func (sf *syncerFixture) Start(t *testing.T) *StartedSyncerFixture {
+	Eventually(t, func() (success bool, reason string) {
+		syncTarget, err := kcpClusterClient.Cluster(syncerConfig.SyncTargetPath).WorkloadV1alpha1().SyncTargets().Get(ctx, syncerConfig.SyncTargetName, metav1.GetOptions{})
+		require.NoError(t, err)
+		if len(syncTarget.Status.VirtualWorkspaces) != 1 {
+			return false, ""
+		}
+		virtualWorkspaceURL = syncTarget.Status.VirtualWorkspaces[0].SyncerURL
+		syncTargetClusterName = logicalcluster.From(syncTarget)
+		return true, "Virtual workspace URL is available"
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "Syncer Virtual Workspace URL not available")
+
+	virtualWorkspaceRawConfig := rawConfig.DeepCopy()
+	virtualWorkspaceRawConfig.Clusters["syncer"] = rawConfig.Clusters["base"].DeepCopy()
+	virtualWorkspaceRawConfig.Clusters["syncer"].Server = virtualWorkspaceURL
+	virtualWorkspaceRawConfig.Contexts["syncer"] = rawConfig.Contexts["base"].DeepCopy()
+	virtualWorkspaceRawConfig.Contexts["syncer"].Cluster = "syncer"
+	virtualWorkspaceRawConfig.Clusters["upsyncer"] = rawConfig.Clusters["base"].DeepCopy()
+	virtualWorkspaceRawConfig.Clusters["upsyncer"].Server = strings.Replace(virtualWorkspaceURL, "/services/syncer/", "/services/upsyncer/", 1)
+	virtualWorkspaceRawConfig.Contexts["upsyncer"] = rawConfig.Contexts["base"].DeepCopy()
+	virtualWorkspaceRawConfig.Contexts["upsyncer"].Cluster = "upsyncer"
+	syncerVWConfig, err := clientcmd.NewNonInteractiveClientConfig(*virtualWorkspaceRawConfig, "syncer", nil, nil).ClientConfig()
+	require.NoError(t, err)
+	syncerVWConfig = rest.AddUserAgent(rest.CopyConfig(syncerVWConfig), t.Name())
+	require.NoError(t, err)
+	upsyncerVWConfig, err := clientcmd.NewNonInteractiveClientConfig(*virtualWorkspaceRawConfig, "upsyncer", nil, nil).ClientConfig()
+	require.NoError(t, err)
+	upsyncerVWConfig = rest.AddUserAgent(rest.CopyConfig(upsyncerVWConfig), t.Name())
+	require.NoError(t, err)
+
+	startedSyncer := &StartedSyncerFixture{
+		SyncerConfig:          syncerConfig,
+		SyncerID:              syncerID,
+		SyncTargetClusterName: syncTargetClusterName,
+		DownstreamConfig:      downstreamConfig,
+		DownstreamKubeClient:  downstreamKubeClient,
+
+		SyncerVirtualWorkspaceConfig:   syncerVWConfig,
+		UpsyncerVirtualWorkspaceConfig: upsyncerVWConfig,
+	}
+
+	// The sync target becoming ready indicates the syncer is healthy and has
+	// successfully sent a heartbeat to kcp.
+	startedSyncer.WaitForClusterReady(ctx, t)
+
+	go wait.UntilWithContext(ctx, func(ctx context.Context) {
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			syncTarget, err := kcpClusterClient.Cluster(syncerConfig.SyncTargetPath).WorkloadV1alpha1().SyncTargets().Get(ctx, syncerConfig.SyncTargetName, metav1.GetOptions{})
+			require.NoError(t, err)
+
+			authChecker := resourcesync.GVRAuthorizationChecker{
+				DownstreamKubeClient: downstreamKubeClient,
+				SyncTargetClient:     kcpSyncTargetClient.WorkloadV1alpha1().SyncTargets(),
+			}
```

Comment:
why does this logic matter for a generic fake syncer fixture? Just set the right condition in the SyncTarget statically and be done with it. No need to simulate permission checks.

Reply:
instead have a very dedicated e2e test that creates a syncer (or multiple) only testing this permission logic.

Reply:
You're right.
I removed this from the PR.
The tests that depend on this still use the full Syncer, as most of themstill expect the resources to be effectively synced.